tests/BrowserTest.php

```diff
@@ -503,4 +503,114 @@ public function testCancelGetRequestShouldCancelUnderlyingSocketConnection()
         $promise = $this->browser->get('http://example.com/');
         $promise->cancel();
     }
+
+    public function testWithElseHeader()
```

Comment:
Can we give this test a more descriptive name? It looks like this might already be covered by the next test?